models/issue_comment.go

```diff
@@ -568,47 +574,68 @@ func GetCommentByID(id int64) (*Comment, error) {
 	return c, nil
 }
 
-func getCommentsByIssueIDSince(e Engine, issueID, since int64) ([]*Comment, error) {
-	comments := make([]*Comment, 0, 10)
-	sess := e.
-		Where("issue_id = ?", issueID).
-		Where("type = ?", CommentTypeComment).
-		Asc("created_unix")
-	if since > 0 {
-		sess.And("updated_unix >= ?", since)
+// FindCommentsOptions describes the condtions to Find comments
+type FindCommentsOptions struct {
+	RepoID  int64
+	IssueID int64
+	Since   int64
+	Type    CommentType
+}
+
+func (opts *FindCommentsOptions) toConds() builder.Cond {
+	var cond = builder.NewCond()
+	if opts.RepoID > 0 {
+		cond = cond.And(builder.Eq{"issue.repo_id": opts.RepoID})
+	}
+	if opts.IssueID > 0 {
+		cond = cond.And(builder.Eq{"issue.id": opts.IssueID})
 	}
-	return comments, sess.Find(&comments)
+	if opts.Since > 0 {
+		cond = cond.And(builder.Gte{"comment.updated_unix": opts.Since})
+	}
+	if opts.Type != CommentTypeUnknown {
+		cond = cond.And(builder.Eq{"comment.type": opts.Type})
+	}
+	return cond
 }
 
-func getCommentsByRepoIDSince(e Engine, repoID, since int64) ([]*Comment, error) {
+func findComments(e Engine, opts FindCommentsOptions) ([]*Comment, error) {
 	comments := make([]*Comment, 0, 10)
-	sess := e.Where("issue.repo_id = ?", repoID).
-		Where("comment.type = ?", CommentTypeComment).
-		Join("INNER", "issue", "issue.id = comment.issue_id").
-		Asc("comment.created_unix")
-	if since > 0 {
-		sess.And("comment.updated_unix >= ?", since)
-	}
-	return comments, sess.Find(&comments)
+	return comments, e.Join("INNER", "issue", "issue.id = comment.issue_id").
```

Comment:
The join is only necessary if `opts.RepoID > 0`, it would be preferable to only join when we have to.